app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaScreen.kt

```diff
@@ -66,33 +66,38 @@ import kotlinx.coroutines.flow.SharedFlow
 
 @Composable
 fun ImportMediaScreen(
-    importMediaViewModel: ImportMediaViewModel = hiltViewModel(),
+    importMediaViewModel: ImportMediaUnauthenticatedViewModel = hiltViewModel(),
     featureFlagNotificationViewModel: FeatureFlagNotificationViewModel = hiltViewModel()
 ) {
-    checkIfSharingIsEnabled(featureFlagNotificationViewModel)
-    val context = LocalContext.current
-    LaunchedEffect(importMediaViewModel.importMediaState.importedAssets) {
-        if (importMediaViewModel.importMediaState.importedAssets.isEmpty()) {
-            context.getActivity()?.let { importMediaViewModel.handleReceivedDataFromSharingIntent(it) }
-        }
-    }
+    featureFlagNotificationViewModel.loadInitialSync()
 
     ImportMediaContent(importMediaViewModel, featureFlagNotificationViewModel)
 }
 
 @Composable
-fun checkIfSharingIsEnabled(featureFlagNotificationViewModel: FeatureFlagNotificationViewModel) {
-    LaunchedEffect(Unit) {
-        featureFlagNotificationViewModel.loadInitialSync()
+fun ImportMediaContent(
+    unauthorizedViewModel: ImportMediaUnauthenticatedViewModel,
+    featureFlagNotificationViewModel: FeatureFlagNotificationViewModel
+) {
+
+    when (val state = featureFlagNotificationViewModel.featureFlagState.fileSharingRestrictedState) {
+        FeatureFlagState.SharingRestrictedState.NO_USER -> ImportMediaLoggedOutContent(
```

Comment:
By default in `FeatureFlagState` it's `NONE` and it can try to create a screen with value `NONE` before it will be executed and updated. I actually tried and I get `java.lang.IllegalStateException: no current session was found` when I'm logged out and try to share an image with Wire. It instantly tries to create a composable with `ImportMediaRegularContent()` and throws an error because `ImportMediaAuthenticatedViewModel` can't be initialized.

Reply:
true story. 
Added nullability to `fileSharingRestrictedState` so Compose will wait till some value comes